pkg/dt/node.go

```diff
@@ -97,6 +98,39 @@ func (n *Node) LookProperty(name string) (*Property, bool) {
 	return nil, false
 }
 
+// PropertyBytes returns the bytes for a property of a Node, or an error
+// if there is no such property.
+func (n *Node) PropertyBytes(property string) ([]byte, error) {
```

Comment:
What do you think of something like this to propagate the error value:

```
func (fdt *FDT) Query() NodeQuery {
    return NodeQuery{n: fdt.RootNode}
}

type NodeQuery struct {
    n *Node
    err error
}

func (nq *NodeQuery) Node(name string) NodeQuery {
    if nq.err != nil {
        return nq
    }
    for _, n := nq.n.Children {
        if n.Name == name {
            return NodeQuery{n: n}
        }
    }
    return NodeQuery{err: fmt.Errorf("cannot find node name %q", name)}
}

func (nq *NodeQuery) Property(name string) PropertyQuery {
    if nq.err != nil {
        return PropertyQuery{err: nq.err}
    }
    for _, p := nq.n.Properties {
        if p.Name == name {
            return PropertyQuery{p: p}
        }
    }
    return PropertyQuery{err: fmt.Errorf("cannot find property name %q", name)}
}

type PropertyQuery struct {
    p *Property
    err error
}

func (pq *PropertyQuery) AsU32() (uint32, error) {
    if pq.err != nil {
        return 0, pq.err
    }
    return pq.p.AsU32(), nil
}
```

Then to access the value `/serial/uart0/baud_rate`, the error is propagated through:

```
baud, err := fdt.Query().Node("serial").Node("uart0").Property("baud_rate").AsU32();
```